components/__tests__/UnitInput.test.ts

```diff
@@ -0,0 +1,163 @@
+
+    expect(inputEventValue).toBe('2G');
+
+    wrapper.setProps({ outputModifier: false });
+    await wrapper.vm.$nextTick();
+    inputWrapper.setValue(3);
+    inputEventValue = wrapper.emitted('input')[1][0];
+
+    expect(inputEventValue).toBe(3000000000);
+  });
+
+  test('Output si modifier matches displayed units', () => {
+    const wrapper = mount(UnitInput, {
+      propsData: {
+        value: 1, inputExponent: 3, outputModifier: true
+      }
+    });
+    const inputWrapper = wrapper.find('input');
+
+    inputWrapper.setValue(2);
+    const inputEventValue = wrapper.emitted('input')[0][0];
+    const inputEventUnitModifier = inputEventValue.split(/[0-9]/)[1];
+
+    const addonText = wrapper.find('.addon').text();
+    const addonUnitModifier = addonText.slice(0, -1);
+
+    expect(inputEventUnitModifier).toEqual(addonUnitModifier);
+  });
+
+  test('Prop increment=1024 changes si base unit (MB to MiB)', async() => {
+    const wrapper = mount(UnitInput, {
+      propsData: {
+        value: 1, inputExponent: 2, increment: 1024
+      }
+    });
+    let addonText = wrapper.find('.addon').text();
+    let addonUnitModifier = addonText.slice(0, -1);
+
+    expect(addonUnitModifier).toBe('Mi');
+
+    wrapper.setProps({ increment: 1000 });
+    await wrapper.vm.$nextTick();
+
+    addonText = wrapper.find('.addon').text();
+    addonUnitModifier = addonText.slice(0, -1);
+    expect(addonUnitModifier).toBe('M');
+  });
+
+  test('Prop outputAs toggles string/number output', async() => {
+    const wrapper = mount(UnitInput, {
+      propsData: {
+        value: 1, inputExponent: 3, outputAs: 'string'
+      }
+    });
+    const inputWrapper = wrapper.find('input');
+
+    inputWrapper.setValue(2);
+    let inputEventValue = wrapper.emitted('input')[0][0];
+
+    expect(typeof inputEventValue).toBe('string');
+
+    wrapper.setProps({ outputAs: 'number' });
+    await wrapper.vm.$nextTick();
+    inputWrapper.setValue(1);
+    inputEventValue = wrapper.emitted('input')[1][0];
+
+    expect(typeof inputEventValue).toBe('number');
+  });
+
+  test('Prop baseUnit controls text appended to si modifier', async() => {
+    const wrapper = mount(UnitInput, { propsData: { value: 1, inputExponent: 2 } });
+    let addonText = wrapper.find('.addon').text();
+
+    expect(addonText).toBe('MB');
+
+    wrapper.setProps({ baseUnit: 'Coolio' });
+    await wrapper.vm.$nextTick();
+    addonText = wrapper.find('.addon').text();
+
+    expect(addonText).toBe('MCoolio');
+  });
+
+  test('Prop suffix sets a display-only suffix', () => {
+    const wrapper = mount(UnitInput, { propsData: { value: 1, suffix: 'seconds' } });
+    const addonText = wrapper.find('.addon').text();
+
+    expect(addonText).toBe('seconds');
+
+    const inputWrapper = wrapper.find('input');
+
+    inputWrapper.setValue(1);
+    const inputEventValue = wrapper.emitted('input')[0][0];
+
+    expect(inputEventValue).toBe(1);
```

Comment:
I do not think that we should test the implementation, but rather the behavior of the component.

Reply:
Given the structure of this component, it seems quite impossible to work exclusively with the behaviors, as the interested value is actually not displayed.

I set this as resolved and we may eventually discuss this in another circumstances.